src/main/java/net/glowstone/inventory/GlowPlayerInventory.java

```diff
@@ -378,17 +393,24 @@ public Location getLocation() {
         return owner.getLocation();
     }
 
+    /**
+     * Remove all matching items from the inventory.
+     *
+     * @param type the item to remove, or null to remove everything
+     * @param data the data value to match on, or null to match all data values
+     * @return the number of items (not stacks) removed
+     */
     public int clear(Material type, MaterialData data) {
+        if (type == Material.AIR) {
+            return 0;
+        }
         int numCleared = 0;
         for (int i = 0; i < getSize(); ++i) {
             ItemStack stack = getItem(i);
             if (stack != null && (type == null || stack.getType() == type) && (data == null || stack
                 .getData().equals(data))) {
                 setItem(i, InventoryUtil.createEmptyStack());
-                if (!InventoryUtil.isEmpty(stack)) {
```

Comment:
This check still needs to be here in the case of type being null.

Reply:
Reverted.